### PR TITLE
Revert "Add an otelcol.processor.attributes component (#3349)"

### PR DESCRIPTION
**Pull request by ptodev**
Removing `otelcol.processor.attributes` for now, because it does not seem to work with all signal types.

The [Test_ExcludeMulti](https://github.com/grafana/agent/blob/more_unit_tests_for_otelcol_processor_attributes/component/otelcol/processor/attributes/attributes_test.go#L489) test fails with this error:
```
services, span_names, and span_kinds are not valid for log records
```
The issue is that we [pass](https://github.com/grafana/agent/blob/main/component/otelcol/processor/processor.go#L178) the logs processor incompatible match properties. [ValidateForLogs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.63.0/internal/coreinternal/processor/filterconfig/config.go#L188) complains because there are "services" specified, which otel doesn't expect for logs.

### CODE REVIEW

File: docs/sources/flow/reference/components/discovery.kubernetes.md [hunk 1]
```diff
@@ -269,7 +269,7 @@ omitted, all namespaces are searched.
 Name | Type | Description | Default | Required
 ---- | ---- | ----------- | ------- | --------
 `own_namespace` | `bool`   | Include the namespace the agent is running in. | | no
-`names` | `list(string)` | List of namespaces to search. | | no
+`names` | `[]string` | List of namespaces to search. | | no
```
Comment (spartan0x117) [inline docs/sources/flow/reference/components/discovery.kubernetes.md:272]: Did we want to revert this change as well?

Suggestion in case you want to apply from GH:
```suggestion
`names` | `list(string)` | List of namespaces to search. | | no
```

File: docs/sources/flow/reference/components/mimir.rules.kubernetes.md [hunk 2]
```diff
@@ -120,11 +120,11 @@ The `match_expression` block describes a Kubernetes label match expression for r
...
-Name       | Type           | Description                                        | Default | Required
------------|----------------|----------------------------------------------------|---------|---------
-`key`      | `string`       | The label name to match against.                   |         | yes
-`operator` | `string`       | The operator to use when matching.                 |         | yes
-`values`   | `list(string)` | The values used when matching.                     |         | no
+Name       | Type       | Description                                        | Default | Required
+-----------|------------|----------------------------------------------------|---------|---------
+`key`      | `string`   | The label name to match against.                   |         | yes
+`operator` | `string`   | The operator to use when matching. |         | yes
+`values`   | `[]string` | The values used when matching.                     |         | no
```
Comment (spartan0x117) [inline docs/sources/flow/reference/components/mimir.rules.kubernetes.md:127]: Did we want to revert this change as well?

Suggestion in case you want to apply from GH:
```suggestion
Name       | Type           | Description                                        | Default | Required
-----------|----------------|----------------------------------------------------|---------|---------
`key`      | `string`       | The label name to match against.                   |         | yes
`operator` | `string`       | The operator to use when matching.                 |         | yes
`values`   | `list(string)` | The values used when matching.                     |         | no
```

**Review by spartan0x117 — APPROVED** [review]
:+1:

File: docs/sources/flow/reference/components/otelcol.processor.batch.md [hunk 3]
```diff
@@ -112,6 +112,7 @@ otelcol.exporter.otlp "production" {
     endpoint = env("OTLP_SERVER_ENDPOINT")
   }
 }
+
```
Comment (ptodev) [inline docs/sources/flow/reference/components/otelcol.processor.batch.md:115]:
```suggestion
```
